components/map/basic/src/index.js

```diff
@@ -0,0 +1,227 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+
+class MapBasic extends Component {
+  constructor (props) {
+    super(props)
+    this.map = this._createMapObject()
+    this._setCurrentLoadedLibraries()
+    this._setCurrentLoadedStyles()
+    this._initLibraryInjection(props.libraries, props.appId, props.appCode, props.useCIT, props.useHTTPS)
+    this._initStylesInjection(props.styles)
+  }
+
+  _createMapObject () {
+    return {
+      instance: null,
+      behavior: null,
+      ui: null,
+      platform: null,
+      defaultLayers: null
+    }
+  }
+
+  _setCurrentLoadedStyles () {
+    this.currentStylesLoaded = Array.from(document.querySelectorAll('link')).map(style => style.href)
+  }
+
+  _setCurrentLoadedLibraries () {
+    this.currentLibrariesLoaded = Array.from(document.querySelectorAll('script')).map((library) => library.src)
+  }
+
+  _buildMapUIComponents () {
+    this.map.ui = window.H.ui.UI.createDefault(this.map.instance, this.map.defaultLayers)
+  }
+
+  _makeInteractive () {
+    this.map.behavior = new window.H.mapevents.Behavior(new window.H.mapevents.MapEvents(this.map.instance))
+  }
+
+  _initStylesInjection (styles) {
+    const stylesLength = styles.length
+    const styleUrl = styles.shift()
+
+    if (!this.currentStylesLoaded.includes(styleUrl) && stylesLength) {
+      const link = document.createElement('link')
+      link.href = styleUrl
+      link.rel = 'stylesheet'
+      link.type = 'text/css'
+      document.body.appendChild(link)
+      this._initStylesInjection(styles)
+    } else if (stylesLength) {
+      this._initStylesInjection(styles)
+    }
+  }
+
+  _initLibraryInjection (libraries, appId, appCode, useCIT, useHTTPS) {
+    const libraryLength = libraries.length
+    const libraryUrl = libraries.shift()
+
+    if (!this.currentLibrariesLoaded.includes(libraryUrl) && libraryLength) {
+      const script = document.createElement('script')
```

Comment:
You are appending it to body. It's a blocking script. Your component is blocking the full page delaying the render to the load of these libraries. The script should be async.